spec/javascripts/components/search-with-autocomplete-spec.js

```diff
@@ -0,0 +1,211 @@
+  }
+
+  const html = `
+    <form id="search-form">
+      <div
+        class="gem-c-search-with-autocomplete"
+        data-module="search-with-autocomplete"
+        data-source="https://www.example.org/api/autocomplete.json?foo=bar"
+        data-source-key="suggestions"
+      >
+        <div
+          class="gem-c-search gem-c-search--separate-label"
+          data-module="gem-toggle-input-class-on-focus"
+        >
+          <label for="search-main-7fdd251c" class="gem-c-search__label">Search on GOV.UK</label>
+          <div class="gem-c-search__item-wrapper">
+            <div class="gem-c-search__input-wrapper">
+              <input
+                enterkeyhint="search"
+                class="gem-c-search__item gem-c-search__input js-class-toggle" id="search-main-7b87262d"
+                name="q"
+                title="Search"
+                type="search"
+                value="i've been looking for freedom"
+              >
+            </div>
+
+            <div class="gem-c-search__item gem-c-search__submit-wrapper">
+              <button class="gem-c-search__submit" type="submit" enterkeyhint="search">
+                Search
+              </button>
+            </div>
+          </div>
+        </div>
+      </div>
+    </form>
+  `
+  const performInput = (input, value, onDone) => {
+    input.value = value
+    input.dispatchEvent(new Event('input'))
+
+    // Gives the autocomplete component a chance to do its work and potentially update the DOM. We
+    // could use a mutation observer for this, but that would only work if there are any results (as
+    // there would be no updates if there are no results).
+    setTimeout(onDone, 5)
+  }
+
+  const expectMenuToBeShownWithOptions = (options) => {
+    const menu = fixture.querySelector('.autocomplete__menu')
+    const results = [...menu.querySelectorAll('.autocomplete__option')].map(
+      (r) => r.textContent.trim()
+    )
+    expect(menu.classList).toContain('autocomplete__menu--visible')
+    expect(results).toEqual(options)
+  }
+
+  const expectMenuNotToBeShown = () => {
+    const menu = fixture.querySelector('.autocomplete__menu')
+    expect(menu.classList).not.toContain('autocomplete__menu--visible')
+  }
+
+  const stubSuccessfulFetch = (suggestions) => {
+    spyOn(window, 'fetch').and.returnValue(Promise.resolve({
+      json: () => Promise.resolve({ suggestions })
+    }))
+  }
+
+  const stubLocalErrorFetch = () => {
+    spyOn(window, 'fetch').and.returnValue(Promise.reject(new Error('Network error')))
+  }
+
+  const stubServerErrorFetch = () => {
+    spyOn(window, 'fetch').and.returnValue(Promise.resolve({
+      ok: false,
+      status: 500,
+      json: () => Promise.resolve({ error: 'Internal server error' })
+    }))
+  }
+
+  beforeEach(() => {
+    loadAutocompleteComponent(html)
+    autocomplete.init()
+  })
+
+  afterEach(() => {
+    fixture.remove()
+  })
+
+  it('deletes the original input', () => {
```

Comment:
Does this test prove that the original input has been deleted? If the JavaScript hadn't run I think there'd still be `1` inputs. Not sure what to suggest - we're basically testing that the input is destroyed then exactly reproduced. Maybe just remove this test?